vyper/semantics/types/function.py

```diff
@@ -478,6 +478,10 @@ def fetch_call_return(self, node: vy_ast.Call) -> Optional[BaseTypeDefinition]:
         for kwarg in node.keywords:
             if kwarg.arg in self.call_site_kwargs:
                 kwarg_settings = self.call_site_kwargs[kwarg.arg]
+                if kwarg_settings.require_return and self.return_type is None:
```

Comment:
probably this branch is cleaner if we don't modify the KwargSettings data structure.
```suggestion
                if kwarg.arg == "default_return_value" and self.return_type is None:
```